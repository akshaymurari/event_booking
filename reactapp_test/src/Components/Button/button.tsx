import React from 'react';

const Button = (props:{buttontext:string,takebuttontext:boolean}) => {
    return (
        <>
            <button data-testid="button">{
                (!props.takebuttontext)?props?.buttontext:"button"
            } hii</button>
        </>
    )
}

export default Button;