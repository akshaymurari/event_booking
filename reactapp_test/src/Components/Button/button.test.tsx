import {render,screen,cleanup} from "@testing-library/react";
import Button from "./button";
import renderer from "react-test-renderer";

afterEach(()=>{
    cleanup();
});

test("button",()=>{
    render(<Button buttontext="hello" takebuttontext={false} />);
    const button = screen.getByTestId("button");
    expect(button.innerHTML).toBe("hello hii");
});

test("button",()=>{
    render(<Button buttontext="hello1" takebuttontext={true}/>);
    const button = screen.getByTestId("button");
    expect(button.innerHTML).toBe("button hii");
});


test("snapshots",()=>{
    const tree = renderer.create(<Button buttontext="hello" takebuttontext={true}></Button>);
    expect(tree).toMatchSnapshot();
});