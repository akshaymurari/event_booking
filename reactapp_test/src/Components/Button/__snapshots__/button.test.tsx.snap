// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots 1`] = `
<button
  data-testid="button"
>
  button
   hii
</button>
`;
